components/bl-credit-card-component/component.json

```diff
@@ -0,0 +1,245 @@
+      "label": "CVC Field Placeholder",
+      "showInSettings": true,
+      "defaultValue": "CVC",
+      "hasLogicHandler": false,
+      "handlerId": "cvcFieldPlaceholderLogic"
+    },
+    {
+      "type": "text",
+      "name": "borderWidth",
+      "label": "Border Width",
+      "showInSettings": true,
+      "hasLogicHandler": false,
+      "handlerId": "borderWidthLogic",
+      "defaultValue": "1px"
+    },
+    {
+      "type": "select",
+      "name": "borderStyle",
+      "label": "Border Style",
+      "showInSettings": true,
+      "hasLogicHandler": false,
+      "handlerId": "borderStyleLogic",
+      "options": [
+        {
+          "value": "none",
+          "label": "none"
+        },
+        {
+          "value": "solid",
+          "label": "solid"
+        },
+        {
+          "value": "dotted",
+          "label": "dotted"
+        },
+        {
+          "value": "dashed",
+          "label": "dashed"
+        },
+        {
+          "value": "double",
+          "label": "double"
+        }
+      ],
+      "defaultValue": "solid"
+    },
+    {
+      "type": "color",
+      "name": "borderColor",
+      "label": "Border Color",
+      "showInSettings": true,
+      "hasLogicHandler": false,
+      "handlerId": "borderColorLogic"
+    }
+  ],
+  "eventHandlers": [
+    {
+      "name": "onSubmit",
+      "label": "On Submit Event",
+      "handlerDescription": "This event is fired when the card form is submitted.",
+      "contextBlocks": [
+        {
+          "id": "cardDetails",
+          "label": "Сard Details"
+        }
+      ]
+    },
+    {
+      "name": "onSuccess",
+      "label": "On Success Event",
+      "contextBlocks": [
+        {
+          "id": "cardDetails",
+          "label": "Сard Details"
+        }
+      ],
+      "handlerDescription": "This event is fired when the payment is complete."
+    },
+    {
+      "name": "onReject",
+      "label": "On Reject Event",
+      "contextBlocks": [
+        {
+          "id": "error",
+          "label": "Error"
+        }
+      ],
+      "handlerDescription": "This event is fired when the payment fails."
+    },
+    {
+      "name": "onFocus",
+      "label": "On Focus Event",
+      "handlerDescription": "This event is fired when the Card Form gains focus.",
+      "contextBlocks": [
+        {
+          "id": "event",
+          "label": "Event"
+        },
+        {
+          "id": "formData",
+          "label": "Form Data"
+        }
+      ]
+    },
+    {
+      "name": "onBlur",
+      "label": "On Blur Event",
+      "handlerDescription": "This event is fired when the Card Form loses focus.",
+      "contextBlocks": [
+        {
+          "id": "event",
+          "label": "Event"
+        },
+        {
+          "id": "formData",
+          "label": "Form Data"
+        }
+      ]
+    }
+  ],
+  "actions": [
+    {
+      "id": "clearForm",
+      "label": "Clear Form of"
+    },
+    {
+      "id": "validateNumber",
```

Comment:
what's the purpose of having separated validators for NNumber/CVC/Expiration?

Reply:
Check validation before submitting the form, for example when the form inputs loses focus.